src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,75 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception when executed. If it does, the exception
+     * object is returned. If it does not, an {@link AssertionError} is thrown.
+     *
+     * @param runnable A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     */
+    public static <T extends Throwable> T expectThrows(ThrowingRunnable runnable) {
+        return (T) expectThrows(null, Throwable.class, runnable);
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception when executed. If it does, the exception
+     * object is returned. If it does not, an {@link AssertionError} is thrown with the given {@code
+     * message}.
+     *
+     * @param message  the identifying message for the {@link AssertionError}
+     * @param runnable A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     */
+    public static <T extends Throwable> T expectThrows(String message, ThrowingRunnable runnable) {
```

Comment:
First of all, you're not objecting to my API design or my testing practices, you're objecting to catching `Throwable`. And you can play similar games with `Exception` too: what are you, crazy? You can't catch `Exception`! What if that's an `InterruptedException` you just swallowed?! You didn't set the interrupted flag, you monster!

Catching `RuntimeException`, if it even works, is just a different kind of hack. In your case, it's not what you actually want to catch, it just happens to be the "most recent common ancestor" of the exceptions that you're actually interested in, but since there's no way to specify a type union in Java you specify an ancestor class instead. I could just as easily respond by saying: don't you care if your constructor throws `ArithmeticException`? You weren't actually expecting the constructor to divide by zero, were you? Now the test is passing for the wrong reason!

Furthermore: no, I don't care if my constructor throws `ThreadDeath` or `StackOverflowError`, because I know apodictically and with absolute metaphysical certitude that no constructor _I_ wrote is going to throw a `VirtualMachineError` during a unit test, unless the entire JVM is coincidentally destabilized at the exact same time for an unrelated reason.

Finally, even if you're very very precise in the types that you throw and catch, and you assert that (say) `NullPointerException` and _only_ `NullPointerException` is caught, you _still_ haven't necessarily demonstrated that you're making correct assertions. You might be catching an NPE thrown by a different (e.g. prior) statement, or one that was propagated from code the SUT called, and then your code is passing for the wrong reason. So at this point we have a few options:
* Associate a unique public exception type with each `throws` statement (which still doesn't work with recursion, maybe we need to add global locking or some sort of `Semaphore`)
* Require the user to specify an assertion containing both the exception type _and_ the message that that exception is to contain, and then I guess publish some sort of Checkstyle config that ensures at build time that all such messages are unique (I guess this brings us to those opaque Microsoftian "error codes")
* Be reasonable and concede that "this call does not return via normal control flow" is a coherent and intelligible assertion that can stand alone in a unit test